net/isoping/isoping_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+package isoping
+
+import (
+	"bytes"
+	"encoding/binary"
+	"math"
+	"testing"
+)
+
+// Tests if our stddev calculation is within reason
+// Must do some rounding to a certain significant digit
+// Currently only need 6 digits for the testing.
+func sigDigs(x float64, digs int) float64 {
+	return math.Round(x*math.Pow10(digs)) / math.Pow10(digs)
+}
+
+// TestOnepass_stddev tests if the function receives the same answer as in
+// the C implementation of this function.
+func TestOnepass_stddev(t *testing.T) {
+	t.Parallel()
+
+	answer := sigDigs(onepass_stddev(12, 2, 3), 6)
+	expected := 2.309401
+	answer2 := sigDigs(onepass_stddev(12023232232, 212, 321), 6)
+	expected2 := 6129.649279
+	if answer != expected {
+		t.Errorf("got %v, expected %v", answer, expected)
+	}
+	if answer2 != expected2 {
+		t.Errorf("got %v, expected %v", answer2, expected2)
+	}
+}
+
+// TestUstimeCast tests if casting was correct
+// sanity check, probably will be removed for redundancy
```

Comment:
Recommend not including the "probably will be removed..." because invariably, three years from now someone will read that and wonder if they should delete it.

"Too many tests" is rarely an actual problem that a codebase has.